### PR TITLE
Add `augerctl` like kubectl

### LINKED ISSUE #3: Migrate the etcd implementations of the operations in kwok to here.
[kwok](https://github.com/kubernetes-sigs/kwok/tree/main/pkg/kwokctl/etcd) is a simulated k8s cluster, and many of its capabilities will be achieved by manipulating etcd directly, which may be more appropriate here.

### PR DESCRIPTION
**Pull request by wzshiming**
Fixes #3

### Preparation cluster
Create a cluster and expose etcd port, to facilitate the creation of a cluster using kwokctl, this can be any other k8s cluster
``` bash
# brew install kwok
kwokctl create cluster --etcd-port 2379
```

### Get a single resource

Get the kubernetes.default service

``` bash
go run ./cmd/augerctl get services -n default kubernetes
```

### Get the all of the resource

``` bash
go run ./cmd/augerctl get leases -n kube-system
```

### Get the all of the etcd
``` bash
go run ./cmd/augerctl get
``` 

### Modify immutable data

``` bash
# change the creation time to very long ago
go run ./cmd/augerctl get services -n default kubernetes | sed 's/creationTimestamp: .*/creationTimestamp: "2006-01-02T15:04:05Z"/' | go run ./cmd/augerctl put --path -
kubectl get services -n default kubernetes
```

> Maybe patch subcommands can be added in the future

### Delete data
``` bash
go run ./cmd/augerctl del services -n default kubernetes
```

### CODE REVIEW

File: cmd/ctl/main.go
Comment (wzshiming) [inline cmd/ctl/main.go:1]: Added to cmd as a subcommand, run an error.
So put it here temporarily to verify the feature.

```log
panic: proto: duplicate enum registered: mvccpb.Event_EventType

goroutine 1 [running]:
github.com/golang/protobuf/proto.RegisterEnum({0x105609d6b, 0x16}, 0x140001cd410?, 0x140001cd410)
        /Users/zsm/go/pkg/mod/github.com/golang/protobuf@v1.5.4/proto/registry.go:98 +0xb4
go.etcd.io/etcd/api/v3/mvccpb.init.0()
        /Users/zsm/go/pkg/mod/go.etcd.io/etcd/api/v3@v3.5.13/mvccpb/kv.pb.go:159 +0x38
exit status 2
```

File: pkg/ctl/command/get/get.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+func runE(ctx context.Context, etcdclient client.Client, flags *flagpole, args []string) error {
+	var targetGr schema.GroupResource
+	var targetName string
+	var targetNamespace string
+	if len(args) != 0 {
+		// TODO: Support get info from scheme.Codecs and CRD
+		//       Support short name
+		//       Check whether there is a namespace
```
Comment (wzshiming) [inline pkg/ctl/command/get/get.go:76]: Plan to do it next time

File: pkg/ctl/command/get/global.go
Comment (wzshiming) [inline pkg/ctl/command/get/global.go:1]: Copy from etcdctl, and made some changes, and this PR will continue to refactor

**Comment by wzshiming** [comment]
/assign @jmhbnz
/assign @siyuanfoundation

This PR is expected to add a subcommand like kubectl to manipulate etcd data, do you guys have time to take a look at it and leave suggestions and comments to guide this forward?

File: pkg/ctl/command/global.go [hunk 2]
```diff
@@ -0,0 +1,380 @@
+// Copyright 2015 The etcd Authors
```
Comment (wzshiming) [inline pkg/ctl/command/global.go:1]: Copy from etcdctl, and made some changes, and this PR will continue to refactor

File: cmd/ctl/main.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+	"github.com/etcd-io/auger/pkg/ctl"
+)
+
+func main() {
+	if err := ctl.RootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
```
Comment (wzshiming) [inline cmd/ctl/main.go:31]: Run an error when added to `cmd` as a subcommand,
So put it here temporarily to verify the feature.

```log
panic: proto: duplicate enum registered: mvccpb.Event_EventType

goroutine 1 [running]:
github.com/golang/protobuf/proto.RegisterEnum({0x105609d6b, 0x16}, 0x140001cd410?, 0x140001cd410)
        /Users/zsm/go/pkg/mod/github.com/golang/protobuf@v1.5.4/proto/registry.go:98 +0xb4
go.etcd.io/etcd/api/v3/mvccpb.init.0()
        /Users/zsm/go/pkg/mod/go.etcd.io/etcd/api/v3@v3.5.13/mvccpb/kv.pb.go:159 +0x38
exit status 2
```

**Comment by wzshiming** [comment]
Hi @jmhbnz @siyuanfoundation

Is it out of scope? 
I'm planning to make this a separate project, would it be OK to call it augerctl or ketctl(Kubernetes ETcd ctl)?

**Comment by wzshiming** [comment]
Ahhh, I don't know how much more this can be split, it's already the smallest possible implementation of the tool, and reducing it would only remove the feature.
I plan to create a new repo for it and make it stable, then propose it to the sig-etcd. WDYT?

**Comment by siyuanfoundation** [comment]
> Ahhh, I don't know how much more this can be split, it's already the smallest possible implementation of the tool. I understand your concern that this merge will mess up the current codebase. So I create a new repo https://github.com/wzshiming/kectl, and make it stable, and then propose it to the sig-etcd. WDYT?

Examples of how to breaking up the PR could be 
1. 1 PR for `get`
2. 1 PR for `delete`
3. ...


Yes, I think the idea could be worth discussing in sig-etcd meeting.

**Comment by wzshiming** [comment]
> > Ahhh, I don't know how much more this can be split, it's already the smallest possible implementation of the tool. I understand your concern that this merge will mess up the current codebase. So I create a new repo https://github.com/wzshiming/kectl, and make it stable, and then propose it to the sig-etcd. WDYT?
> 
> Examples of how to breaking up the PR could be
> 
> 1. 1 PR for `get`
> 2. 1 PR for `delete`
> 3. ...


I've created a new PR for the `get` and removed optional features. #82

**Comment by jmhbnz** [comment]
Closed in favor of #82

**Pull request closed without merging**